### DISCUSSION TITLE
How to re-render DeckGL layers via layer instance?

### DISCUSSION THREAD

**Discussion by vivek091195** [visgl/deck.gl#6320] [Q&A]
I am using DeckGL over Google maps in my project. I am rendering layers like this,
`  const layer = new GeoJsonLayer({
    id: 'earthquakes',
    data: 'https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_month.geojson',
    filled: true,
    pointRadiusMinPixels: 2,
    pointRadiusMaxPixels: 200,
    opacity: 0.4,
    pointRadiusScale: 0.3,
    getRadius: (f: Feature) => Math.pow(10, f.properties.mag),
    getFillColor: [255, 70, 30, 180],
    autoHighlight: true,
    transitions: {
      getRadius: {
        type: 'spring',
        stiffness: 0.1,
        damping: 0.15,
        enter: () => [0], // grow from size 0,
        duration: 10000,
      },
    },
    onDataLoad: () => {
      /* eslint-disable no-undef */
      // @ts-ignore defined in include
      progress.done(); // hides progress bar
      /* eslint-enable no-undef */
    },
  });

  const deckOverlay = new GoogleMapsOverlay({
    layers: [layer],
  });`

In case the data fetched from 'https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_month.geojson' changes, then is there a way to re-render this layer if I have access to just _deckOverlay_ variable?

**Comment by Pessimistress** [comment]
[setProps](https://deck.gl/docs/api-reference/google-maps/google-maps-overlay#setprops)

**Reply by vivek091195** [reply]
Hi @Pessimistress ,  thanks for the response.

_setProps_ will need an entire layer configuration if I am not wrong. What I want is, if I have 5 layers plotted, then is there a way that I can re-render a few among those layers(not all 5) depending on the layer id? What I am getting as input is the layer id and modified data, and I want to replace the existing layer data with modified data.

**Reply by Pessimistress** [reply]
Deck.gl's design follows the reactive programming paradigm. There is no imperative API to update props directly from a layer instance. You need to do something like this:

```js
let layers = [layer1, layer2, ...];

function updateLayer(id, data) {
  const i = layers.findIndex(l => l.id === id);
  if (i >= 0) {
    layers = layers.slice(); // make a shallow copy
    layers[i] = layers[i].clone({data});
    deckOverlay.setProps(layers);
  }
}
```

See https://deck.gl/docs/developer-guide/using-layers#faq
